packages/api/src/web5.ts

```diff
@@ -326,30 +328,37 @@ export class Web5 {
 
       if (registration !== undefined) {
         // If a registration object is passed, we attempt to register the AgentDID and the ConnectedDID with the DWN endpoints provided
-
         const serviceEndpointNodes = techPreview?.dwnEndpoints ?? didCreateOptions?.dwnEndpoints;
-        for (const dwnEndpoint of serviceEndpointNodes) {
-          try {
 
+        // We only want to return the success callback if we successfully register with all DWN endpoints
+        // So we keep track of whether we attempted registration at all
+        let registrationAttempt = false;
```

Comment:
Style nit: consider moving `registrationAttempt` into the try, because no one outside the try uses it.